multicluster/apis/multicluster/v1alpha1/clusterset_types.go

```diff
@@ -21,31 +21,49 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// MemberCluster defines member cluster information.
-type MemberCluster struct {
-	// Identify member cluster in ClusterSet.
+// LeaderClusterInfo specifies information of a leader cluster.
+type LeaderClusterInfo struct {
+	// Identify a leader cluster in the ClusterSet.
 	ClusterID string `json:"clusterID,omitempty"`
-	// API server of the destination cluster.
+	// API server endpoint of the leader cluster. Used only for a member cluster.
+	// e.g. "https://172.18.0.1:6443", "https://example.com:6443".
 	Server string `json:"server,omitempty"`
-	// Secret name to access API server of the member from the leader cluster.
+	// Name of the Secret resource in the member cluster, which stores
+	// the token to access the leader cluster's API server.
 	Secret string `json:"secret,omitempty"`
-	// ServiceAccount used by the member cluster to access into leader cluster.
+	// ServiceAccount in the leader cluster, from which the member cluster's token
+	// is generated. This is an optional field which helps admin to check
+	// which ServiceAccount is used by a member cluster to access the leader cluster.
+	//
+	// DEPRECATED
+	// This field is planned to be removed in the future releases.
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 }
 
 // ClusterSetSpec defines the desired state of ClusterSet.
 type ClusterSetSpec struct {
+	// ClusterSetID identifies the ClusterSet.
+	// +kubebuilder:validation:Required
+	ClusterSetID string `json:"clusterSetID"`
```

Comment:
Can we use ClusterSet resource name as the ClusterSetID?

Reply:
Yeah, I think it should also work, I will refine this.

Reply:
Refined.